The version pin is irrelevant to how we build and deliver setuptools-scm.

https://github.com/pypa/setuptools-scm/issues/1090
https://github.com/pypa/setuptools-scm/commit/c766df10c18c3c5a6b5741e9f372e193412c0f69

--- setuptools_scm-8.3.1/pyproject.toml
+++ setuptools_scm-8.3.1/pyproject.toml
@@ -4,7 +4,7 @@
 build-backend = "_own_version_helper:build_meta"
 requires = [
   "setuptools>=61",
-  'tomli<=2.0.2; python_version < "3.11"',
+  'tomli; python_version < "3.11"',
 ]
 backend-path = [
   ".",
